lua/rocks/operations/init.lua

```diff
@@ -499,20 +513,29 @@ operations.add = function(arg_list, rock_name, version, callback)
                     message = message,
                 })
             end
-            local function report_error(message)
-                progress_handle:report({
-                    title = "Error",
-                    message = message,
-                })
-                progress_handle:cancel()
-            end
             handler(report_progress, report_error)
             vim_schedule_nio_wait(function()
                 fs.write_file(config.config_path, "w", tostring(user_rocks))
                 progress_handle:finish()
             end)
             return
         end
+        ---@type rock_name
+        local rock_name = arg_list[1]
+        --- We can't mutate the arg_list, because we may need it for a recursive add
```

Comment:
```suggestion
        -- We can't mutate the arg_list, because we may need it for a recursive add
```

nitpick: probably shouldn't be a docstring :p